stl/src/filesystem.cpp

```diff
@@ -907,7 +907,14 @@ struct alignas(long long) _Aligned_file_attrs {
     if (_Bitmask_includes(_Flags, _Attribute_tag_info_data)) {
         FILE_ATTRIBUTE_TAG_INFO _Info;
         if (!GetFileInformationByHandleEx(_Handle._Get(), FileAttributeTagInfo, &_Info, sizeof(_Info))) {
-            return __std_win_error{GetLastError()};
+            const DWORD _Error = GetLastError();
+            if (_Error == ERROR_INVALID_PARAMETER) {

```

Comment:
I don't like checking against `ERROR_INVALID_PARAMETER`.

Can we avoid calling this for non-reparse-points (by checking attributes obtained with other parametet)?

Reply:
`GetFileInformationByHandleEx` with `FileBasicInfo`  returns attribute `FILE_ATTRIBUTE_REPARSE_POINT`

https://docs.microsoft.com/en-us/windows/win32/fileio/file-attribute-constants

Reply:
llvm libc++ works in that way: https://github.com/llvm/llvm-project/blob/e356027016c6365b3d8924f54c33e2c63d931492/libcxx/src/filesystem/posix_compat.h#L172-L194

Reply:
I don't know. I'm fine with checking against `ERROR_INVALID_PARAMETER`. I will wait for more opinions...

Reply:
Thanks for fixing this.

I think relying on `ERROR_INVALID_PARAMETER` was not safe as it is observed behavior of specific filesystem driver; this could be different on another filesystem, like exFAT or ReFS, or even with an update of the existing driver (like in Win12).

I understand the performance consideration for saving a system call, but correctness goes first.